### PR TITLE
Adding thalamocortex neurodamus package

### PR DESCRIPTION
**Pull request by ferdonline**
**This PR**
Create package to install a version suitable for multi-circuit Thalamo-Cortex simulations.
User will be able to `spack load neurodamus-thalamocortex`

**Details**
Instead of further customization of variants, this package is so specialized that it was decided to create an entire new one and avoid polluting neocortex. Version os the package will follow that of neocortex, given it's the main driver

### CODE REVIEW

**Comment by ferdonline** [comment]
@pramodk any remarks here? This will be helpful for these mixed circuits

File: var/spack/repos/builtin/packages/neurodamus-thalamocortex/package.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        ('1.3', '3.2.0'),
+    )
+    version_from_model_core_deps(model_core_dep_v)
+
+    mech_name = "thalamocortex"
+
+    @run_before('build_model')
+    def prepare_mods(self):
+        copy_all('mousify/mod', 'mod', make_link)
+        copy_all('thalamus/mod', 'mod', make_link)
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/neurodamus-thalamocortex/package.py:40]: Without much knowledge here - why thalamocortex is using models of mousify? I was thinking of neocortex.

Comment (ferdonline) [reply var/spack/repos/builtin/packages/neurodamus-thalamocortex/package.py:40]: I also don't know why, but in this circuit they need some mods from mousify...

**Review by matz-e — DISMISSED** [review]

**Comment by matz-e** [comment]
Needs a rebase on current `develop`?

**Comment by matz-e** [comment]
Good to close? This should probably be redone with the new packages / `develop`.

**Pull request closed without merging**